source/PlayerInfo.cpp

```diff
@@ -1134,26 +1140,31 @@ void PlayerInfo::AddShip(const shared_ptr<Ship> &ship)
 
 
 // Adds a ship of the given model with the given name to the player's fleet.
-void PlayerInfo::BuyShip(const Ship *model, const string &name)
+void PlayerInfo::BuyShip(const Ship *model, const string &name, const bool isFromShop)
 {
 	if(!model)
 		return;
 
-	int day = date.DaysSinceEpoch();
-	int64_t cost = stockDepreciation.Value(*model, day);
-	if(accounts.Credits() >= cost)
+	const Ship *newShip = isFromShop ? model : StockShip(model);
```

Comment:
we usually dont bother with making in function variables const (dont ask me why tho)

Reply:
~~this file didnt have many consts tho~~
but yeah I'm not sure either tbh